src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java

```diff
@@ -132,8 +141,10 @@ public boolean accept(File dir, String name) {
 
         LOGGER.finest("Found " + gcLogs.length + " matching files in " + parentDirectory.getAbsolutePath());
         for (File gcLog : gcLogs) {
-            LOGGER.finest("Adding '" + gcLog.getName() + "' file");
-            result.add(new UnfilteredFileContent(GCLOGS_BUNDLE_ROOT + "{0}", new String[]{gcLog.getName()}, gcLog));
+            if (GCLOGS_RETENTION_DAYS <= 0 || (gcLog.lastModified() > (System.currentTimeMillis() - TimeUnit.DAYS.toMillis(GCLOGS_RETENTION_DAYS)))) {
```

Comment:
We could also maybe check the timestamp of the last GC log on the file(pulling a tail of the log) so that we dont get false positives on files which were incorrectly modified recently

Reply:
> What do we do here if they do not rotate very often and have some giant file like 40 MB but it is continuing to be written to. Is there a way to break up the file here?

This is something that could apply to any File that we copy during the bundle generation. This should be a new JIRA IMO.

> We could also maybe check the timestamp of the last GC log on the file(pulling a tail of the log) so that we dont get false positives on files which were incorrectly modified recently

This is another - a bit more expensive - option. I am not sure how many lines we need to tail though to catch a _timestamped_ line. Or if this is worth going that far ? Also they the time expression differs depending on JVM opts, .i.e whether `-XX:+PrintGCTimeStamps` is set or not.

Reply:
To my knowledge I've never seen a single giant GC log file - it's always 1+ years of 100MB files that kill us. Most people set the rotation size at 100MB or less. Most people seem to get that if you enable rotation, you need to specify a limit upon which rotation happens.

Reply:
Will be addressed as part of https://issues.jenkins-ci.org/browse/JENKINS-59030